connection.go

```diff
@@ -1431,39 +1430,152 @@ func (c *Conn) handleFrames(
 	}
 	handshakeWasComplete := c.handshakeComplete
 	var handleErr error
+	var frame wire.Frame
+	var err error
+	var l int
+	var frameType wire.FrameType
+
+	wrapError := func(err error) (isAckEliciting, isNonProbing bool, pathChallenge *wire.PathChallengeFrame, _ error) {
+		return false, false, nil, &qerr.TransportError{
+			ErrorCode:    qerr.FrameEncodingError,
+			ErrorMessage: err.Error(),
+		}
+	}
```

Comment:
1. What's the deal with these many return values? I would've expected a `wrapError` function to have the signature `func(error) error`.
2. I'd prefer to not deal with this in the connection at all.

Instead, how about not exporting `wire.ParseStreamFrame`, but instead exposing a `FrameParser.ParseStreamFrame`. This function would then take care to return the correct `qerr.TransportError`. Makes it a lot easier to test, removes some complexity from the `Conn`, and (probably) compiles down to the exact same thing.